pandas/_libs/tslibs/offsets.pyx

```diff
@@ -364,6 +364,25 @@ class _BaseOffset(object):
                 if name not in ['n', 'normalize']}
         return {name: kwds[name] for name in kwds if kwds[name] is not None}
 
+    def __add__(self, other):
+        if getattr(other, "_typ", None) in ["datetimeindex",
+                                            "series", "period"]:
+            # defer to the other class's implementation
+            return other + self
+        try:
+            return self.apply(other)
+        except ApplyTypeError:
+            return NotImplemented
+
+    def __sub__(self, other):
+        if isinstance(other, datetime):
+            raise TypeError('Cannot subtract datetime from offset.')
+        elif type(other) == type(self):
+            return self.__class__(self.n - other.n, normalize=self.normalize,
```

Comment:
can you change to use ``type(self)`` here, I don't think we use the long form anywhere.

Reply:
its used below in `__mul__`.  change it there too while I'm at it?

Reply:
yeah might as well, its a bit more clean.